### ISSUE TITLE
Cannot import components inside a markdown file

### ISSUE THREAD

**Issue by alihassan1133** [nuxt/content#261]
![image](https://user-images.githubusercontent.com/62854082/87386592-dcd14c80-c56e-11ea-9e0c-0b27c9afa1bc.png)

```
<template>
    <div class="prose prose-lg mx-auto">
        <nuxt-content :document="page" />
    </div>
</template>

<script>
    export default {
        async asyncData ({ $content, params }) {
            const page = await $content(params.slug || 'index').fetch()

            return {
                page
            }
        }
    }
</script>
```

```md
---
title: Getting started
description: ''
---

<example></example>
```

### Steps to reproduce
1. Create a component inside the `components/` directory.
2. Try to include the component in a markdown file.

### What is Expected?
See the component on the web page when you navigate to the URL.

### What is actually happening?
Seeing an error in the console (screenshot attached.)

**Comment by HomyeeKing** [comment]
I have a similiar situation, can not load the markdown files 
![image](https://user-images.githubusercontent.com/49113249/87424728-09d53c00-c60f-11ea-8dd2-96b7525201a3.png)

**Comment by benjamincanac** [comment]
Hey @alihassan1133, you need to register the `<example>` component, you can either put your component inside `components/global/` to be imported automatically or import the component in your page.

**Comment by benjamincanac** [comment]
@HomyeeKing It does not seem to be related, consider creating a new issue next time 🙂.

First of all, my advice would be to surround your `$content` call in a try/catch and call the nuxt `error` method so it doesn't crash and displays the error page from `layouts/error.vue`.

```js
async asyncData({ $content, params, error }) {
    let doc
    try {
      doc = await $content(params.slug || 'index').fetch()
    } catch (e) {
      return error({ statusCode: 404, message: 'Page not found' })
    }

    return {
        doc
    }
}
```

As for your error, it would be easier with a reproduction link but are your sure you have an `index.md` file inside `content/` folder?

**Comment by lucpotage** [comment]
The documentation should help you. Everything is explained : https://content.nuxtjs.org/writing#global-components

**Comment by phiter** [comment]
What if I don't want to use a component globally? Suppose I have a single component for a single blog page, how would I do that?

I would have only one markdown file and a vue file, where would I import the vue file?

**Comment by benjamincanac** [comment]
@phiter You have to import the component in the page you use the `nuxt-content` component:

```vue
<script>
import Example from '~/components/example.vue'

export default {
    components: {
         Example
    }
}
</script>
```

**Comment by melnikovdv** [comment]
I have the same problem. 

Imported as usual, it works in code, but when I move declaration to .md it fails with the same error. 

`<my-image src="/img/w1/logo.png"></my-image>`

**Comment by benjamincanac** [comment]
Could you provide reproduction links so we help you out?

Thanks!

**Comment by benjamincanac** [comment]
It seems there is an issue with local components registration in development due to live edit mode.

For now, to solve the issue you can put your components inside `components/global/` using Nuxt 2.13+ and setting `components: true` in your `nuxt.config.js`, see [Vue components](https://content.nuxtjs.org/writing#vue-components) > `Global components`.

You will also be able to disable the live edit feature once #125 is merged.

**Comment by mlazuardy** [comment]
is this resolved ? in nuxt theme docs it can be override by adding

```
{
 components: [
//import components directory and global directory here
    {
      path: 'your custom component dir path',
      global: true
    }
  ]
}
```
